### PR TITLE
JAMES-2586 Update Guide binding Postgres

### PR DESCRIPTION
**Pull request by vttranlina**

### CODE REVIEW

**Review by quantranhong1999 — APPROVED** [review]

File: server/container/guice/mailbox-postgres/src/main/java/org/apache/james/modules/mailbox/PostgresMailboxModule.java [hunk 1]
```diff
@@ -133,6 +135,7 @@ protected void configure() {
         bind(AttachmentMapperFactory.class).to(PostgresMailboxSessionMapperFactory.class);
 
         bind(ReIndexer.class).to(ReIndexerImpl.class);
+        bind(MessageIdReIndexer.class).to(MessageIdReIndexerImpl.class);
```
Comment (Arsnael) [inline server/container/guice/mailbox-postgres/src/main/java/org/apache/james/modules/mailbox/PostgresMailboxModule.java:138]: What was the issue when we were missing this? Just curious :)

**Review by Arsnael — COMMENTED** [review]
Comment description: Guide => Guice

Otherwise LGTM :)

File: server/container/guice/mailbox-postgres/src/main/java/org/apache/james/modules/mailbox/PostgresMailboxModule.java (hunk 1 above)
Comment (vttranlina) [reply server/container/guice/mailbox-postgres/src/main/java/org/apache/james/modules/mailbox/PostgresMailboxModule.java:138]: It is required for web admin MessageRoute

Comment (vttranlina) [reply server/container/guice/mailbox-postgres/src/main/java/org/apache/james/modules/mailbox/PostgresMailboxModule.java:138]: Wait wait
Do we need MessasgeRoute for postgres?
https://james.apache.org/server/manage-webadmin.html#Administrating_Messages
```
This task is only available on top of Guice Cassandra products or Guice JPA products. It is not part of Memory Guice product.
```
This is an old document, not sure it still correct or not with Postgres

Comment (quantranhong1999) [reply server/container/guice/mailbox-postgres/src/main/java/org/apache/james/modules/mailbox/PostgresMailboxModule.java:138]: likely we need e.g. reindex messages in OpenSearch?

Comment (Arsnael) [reply server/container/guice/mailbox-postgres/src/main/java/org/apache/james/modules/mailbox/PostgresMailboxModule.java:138]: As postgres can use Opensearch fo idexing, yes I think we need the task for reindexing :)

**Pull request merged**